packages/simcore-sdk/tests/integration/test_node_ports_v2_r_clone.py

```diff
@@ -0,0 +1,198 @@
+    mocker.patch(
+        "simcore_sdk.node_ports_common.storage_client.update_file_meta_data",
+        side_effect=_TestException,
+    )
+    reload(r_clone)
+    reload(storage_client)
+
+
+@pytest.fixture
+async def client_session(filemanager_cfg: None) -> AsyncIterable[ClientSession]:
+    async with ClientSession() as session:
+        yield session
+
+
+# UTILS
+
+
+@asynccontextmanager
+async def _get_s3_object(
+    r_clone_settings: RCloneSettings, s3_path: str
+) -> AsyncGenerator["aioboto3.resources.factory.s3.Object", None]:
+    session = aioboto3.Session(
+        aws_access_key_id=r_clone_settings.R_CLONE_S3.S3_ACCESS_KEY,
+        aws_secret_access_key=r_clone_settings.R_CLONE_S3.S3_SECRET_KEY,
+    )
+    async with session.resource(
+        "s3", endpoint_url=r_clone_settings.R_CLONE_S3.endpoint
+    ) as s3:
+        s3_object = await s3.Object(
+            bucket_name=r_clone_settings.R_CLONE_S3.S3_BUCKET_NAME,
+            key=s3_path.lstrip(r_clone_settings.R_CLONE_S3.S3_BUCKET_NAME),
+        )
+        yield s3_object
+
+
+async def _download_s3_object(
+    r_clone_settings: RCloneSettings, s3_path: str, local_path: Path
+):
+    async with _get_s3_object(r_clone_settings, s3_path) as s3_object:
+        await s3_object.download_file(f"{local_path}")
+
+
+def _is_file_present(postgres_db: sa.engine.Engine, s3_object: str) -> bool:
+    with postgres_db.begin() as conn:
+        result = conn.execute(
+            file_meta_data.select().where(file_meta_data.c.file_uuid == s3_object)
+        )
+        result_list = list(result)
+        result_len = len(result_list)
+    assert result_len <= 1, result_list
+    return result_len == 1
+
+
+# TESTS
+
+
+async def test_sync_local_to_s3(
+    r_clone_settings: RCloneSettings,
+    s3_object: str,
+    file_to_upload: Path,
+    local_file_for_download: Path,
+    user_id: int,
+    postgres_db: sa.engine.Engine,
+    client_session: ClientSession,
+    cleanup_s3: None,
+) -> None:
+
+    etag = await r_clone.sync_local_to_s3(
+        session=client_session,
+        r_clone_settings=r_clone_settings,
+        s3_object=s3_object,
+        local_file_path=file_to_upload,
+        user_id=user_id,
+    )
+
+    assert isinstance(etag, str)
+    assert '"' not in etag
+
+    await _download_s3_object(
+        r_clone_settings=r_clone_settings,
+        s3_path=s3_object,
+        local_path=local_file_for_download,
+    )
+
+    # check same file contents after upload and download
+    assert file_to_upload.read_text() == local_file_for_download.read_text()
+
+    assert _is_file_present(postgres_db=postgres_db, s3_object=s3_object) is True
+
+
+async def test_sync_local_to_s3_cleanup_on_error(
+    r_clone_settings: RCloneSettings,
+    s3_object: str,
+    file_to_upload: Path,
+    user_id: int,
+    postgres_db: sa.engine.Engine,
+    client_session: ClientSession,
+    cleanup_s3: None,
+    mock_update_file_meta_data: None,
+) -> None:
```

Comment:
This test should be generalized to both with or without RClone

Reply:
Not sure I understand why this should be the case.